__Q2__

Let us first formulate null hypothesis and alternative hypothesis. 

$H_0$ : $p$ = $0.03$ 

$H_1$ : $p$ $\neq$ $0.03$

a) n = 50

Our test statistic is C = 0.6 * 50 = 3

In [2]:
import scipy

p_value = scipy.stats.binomtest(3, 50, p=0.03).pvalue

print(p_value)

0.18920192463027882


At the 5% significance level, we fail to reject the hypothesis.

b) Now let us repeat the proportion test for n = 350. For this larger sample size, we can use the asymptotic proportion test with the standardized test statistic

Z = ($\hat{p}$ - $p_0$) / $\sqrt{(p_0 * (1-p_0)) / n}$

Evaluating this results in: 

In [3]:
import math

z = (0.06 - 0.03) / math.sqrt((0.03 * 0.97) / 350)

print(z)

3.290097607676896


The critical z-values for the two-tailed test at the 5% significance level are $\pm$ 1.96. Since 3.29 is larger than that value, we reject the null hypothesis for n = 350.

To sum up; for n = 50 we fail to reject the null hypothesis that 3% of the products are faulty. For n = 350, we reject the null hypothesis, given that the observed proportion of faulty products in the sample is still 6%. 